amethyst_renderer/src/pipe/stage.rs

```diff
@@ -0,0 +1,100 @@
+//! A stage in the rendering pipeline.
+
+use error::{Error, Result};
+use pipe::{Target, Targets};
+use pipe::pass::{Pass, PassBuilder};
+use scene::{Model, Scene};
+use std::sync::Arc;
+use types::{Encoder, Factory};
+
+/// A stage in the rendering pipeline.
+#[derive(Clone, Debug)]
+pub struct Stage {
+    enabled: bool,
+    passes: Vec<Pass>,
```

Comment:
Shouldn't be passes split into three kinds?
Pre-processing passes
Main-phase passes that handles models
Post-processing passes

While pre-processing passes should be executed first post-processing passes - last
And only main-phase passes should be executed in parallel

encoder flushing should preserve the order between those kinds